Packs/CiscoUmbrellaReporting/Integrations/CiscoUmbrellaReporting/CiscoUmbrellaReporting.py

```diff
@@ -0,0 +1,1138 @@
+import requests
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+# CORTEX XSOAR COMMAND CONSTANTS
+COUNT = 0
+LIMIT = 50
+FROM_DATE = "-7days"
+TO_DATE = "now"
+OFFSET = 0
+INTEGRATION_CONTEXT_NAME = 'UmbrellaReporting'
+TOKEN_ENDPOINT = "https://management.api.umbrella.com/auth/v2/oauth2/token"
+IP_PARAM = 'ip'
+DOMAIN_PARAM = 'domains'
+SHA256_PARAM = 'sha256'
+INTRUSION_ACTION = 'intrusion_action'
+DATE_TIME_FORMAT = "%b %d, %Y %H:%M %p"
+PAGE_NUMBER_ERROR_MSG = 'Invalid input Error: page number should be greater ' \
+                        'than zero.'
+
+ACTIVITY_TRAFFIC_TYPE_DICT = {
+    "dns": ["limit", "from", "to", "offset", "domains", "ip", "verdict",
+            "threats", "threat_types", "identity_types", "page", "page_size"],
+    "proxy": ["limit", "from", "to", "offset", "domains", "ip", "verdict",
+              "threats", "threat_types", "urls", "ports", "identity_types",
+              "file_name", "amp_disposition", "page", "page_size"],
+    "firewall": ["limit", "from", "to", "offset", "ip", "ports", "verdict",
+                 "page", "page_size"],
+    "intrusion": ["limit", "from", "to", "offset", "ip", "ports",
+                  "signatures", "intrusion_action", "page", "page_size"],
+    "ip": ["limit", "from", "to", "offset", "ip", "ports", "identity_types",
+           "verdict", "page", "page_size"],
+    "amp": ["limit", "from", "to", "offset", "amp_disposition", "sha256",
+            "page", "page_size"]
+}
+
+SUMMARY_TYPE_DICT = {
+    "all": ["limit", "from", "to", "offset", "domains", "urls", "ip",
+            "identity_types", "verdict", "file_name", "threats",
+            "threat_types", "amp_disposition", "page", "page_size", "ports"],
+    "category": ["limit", "from", "to", "offset", "domains", "urls", "ip",
+                 "identity_types", "verdict", "file_name", "threats",
+                 "threat_types", "amp_disposition", "page", "page_size"],
+    "destination": ["limit", "from", "to", "offset", "domains", "urls", "ip",
+                    "identity_types", "verdict", "file_name", "threats",
+                    "threat_types", "amp_disposition", "page", "page_size"],
+    "intrusion_rule": ["limit", "from", "to", "offset", "signatures", "ip",
+                       "identity_types", "intrusion_action", "ports", "page",
+                       "page_size"]
+}
+
+
+class Client(BaseClient):
+    def __init__(self, base_url: str, organisation_id: str,
+                 secret_key: str, client_key: str,
+                 verify=None,
+                 proxy=None):
+        BaseClient.__init__(
+            self,
+            base_url,
+            verify=verify,
+            proxy=proxy,
+            ok_codes=(200,),
+        )
+        self.token_url = TOKEN_ENDPOINT
+        self.secret_key = secret_key
+        self.client_key = client_key
+        self.organisation_id = int(organisation_id) if \
+            organisation_id.isdigit() else organisation_id
```

Comment:
I'm not sure I understood this logic - is the organisation_id should be an int or a string?

Reply:
modified this

Reply:
@RamboV -   I can see that you added the following check to the test module: 
```        if not self.organisation_id.isdigit():
            raise DemistoException("Invalid Input Error:The Organization ID is "
                                   "must be a number")
```
Please move the check to the main function, or to the client constructor so that this exception will be raised for any command, and only for the test module.

Reply:
@RamboV - Please change that.

Reply:
@ShacharKidor Added it to constructor

Reply:
Good :)